core/indexer/elasticsearch.go

```diff
@@ -86,43 +96,57 @@ func (ei *elasticIndexer) SaveBlock(
 		return
 	}
 
-	go ei.database.SaveHeader(headerHandler, signersIndexes)
+	go ei.database.SaveHeader(headerHandler, signersIndexes, body, notarizedHeadersHashes)
 
 	if len(body.MiniBlocks) == 0 {
 		log.Debug("indexer", "error", ErrNoMiniblocks.Error())
 		return
 	}
 
-	if ei.options.TxIndexingEnabled {
-		go ei.database.SaveTransactions(body, headerHandler, txPool, ei.shardCoordinator.SelfId())
-	}
-}
+	go ei.database.SaveMiniblocks(headerHandler, body)
```

Comment:
would it worth to add also a flag condition for indexing miniblocks ? maybe add a TODO or wait for the second/third reviewer